frontend2/src/api/helpers.ts

```diff
@@ -84,3 +86,42 @@ export const prefetchNextPage = async <
     }
   }
 };
+
+const safeLoaderHelper = async <T, K>(
+  request: T,
+  factory: QueryFactory<T, K> | PaginatedQueryFactory<T, K>,
+  queryClient: QueryClient,
+): Promise<void> => {
+  try {
+    await queryClient.ensureQueryData({
+      queryKey: buildKey(factory.queryKey, request),
+      queryFn: async () => await factory.queryFn(request, queryClient, true),
```

Comment:
I think there is some TS trickery going on here... this function signature matches `PaginatedQueryFuncBuilder` but works when called on that or `QueryFuncBuilder` ??? No idea how but it seems to work flawlessly.

Reply:
> In JavaScript, if you call a function with more arguments than there are parameters, the extra arguments are simply ignored. TypeScript behaves the same way. Functions with fewer parameters (of the same types) can always take the place of functions with more parameters.

https://www.typescriptlang.org/docs/handbook/2/functions.html